### ISSUE TITLE
Foxx: job failure bug?

### ISSUE THREAD

**Issue by bangert** [arangodb/arangodb#2042]
## my environment running ArangoDB

I'm using the latest ArangoDB of the respective release series:
- [ ] 2.8
- [x] 3.0
- [ ] self-compiled devel branch

On this operating system:
- [ ] DCOS on
  - [ ] AWS
  - [ ] Azure
  - [ ] own infrastructure
- [ ] Linux 
  - [ ] Debian .deb
  - [ ] Ubuntu .deb
  - [ ] SUSE   .rpm
  - [ ] RedHat .rpm
  - [x] Fedora .rpm
  - [ ] Gentoo
  - [ ] docker - official docker library
  - [ ] other:
- [ ] Windows, version: 
- [ ] MacOS, version:
### Foxx

This is arangodb-3.0.5.
In a foxx queue, i cant seem to get error handling to work.

upon failure i see the following in the log:

2016-09-08T08:03:25Z [1003] INFO PENDING: 
2016-09-08T08:03:25Z [1003] INFO [ "_jobs/7643852" ]
2016-09-08T08:03:25Z [1003] INFO FAILED: 
2016-09-08T08:03:25Z [1003] INFO [ ]
2016-09-08T08:03:25Z [1003] INFO COMPLETE: 
2016-09-08T08:03:25Z [1003] INFO [ ]
2016-09-08T08:03:25Z [1003] INFO PROGRESS: 
2016-09-08T08:03:25Z [1003] INFO [ ]
2016-09-08T08:03:26Z [1003] ERROR Job 7643852 failed:
2016-09-08T08:03:26Z [1003] ERROR ArangoError: failed to invoke module
2016-09-08T08:03:26Z [1003] ERROR File: /var/lib/arangodb3-apps/_db/mikeno/mikeno/APP/scripts/publish-to-subscribers.js
2016-09-08T08:03:26Z [1003] ERROR Cause: AssertionError: HTTP returned a failure
<stack trace removed>
2016-09-08T08:03:39Z [1003] INFO PENDING: 
2016-09-08T08:03:39Z [1003] INFO [ "_jobs/7643922" ]
2016-09-08T08:03:39Z [1003] INFO FAILED: 
2016-09-08T08:03:39Z [1003] INFO [ ]
2016-09-08T08:03:39Z [1003] INFO COMPLETE: 
2016-09-08T08:03:39Z [1003] INFO [ ]
2016-09-08T08:03:39Z [1003] INFO PROGRESS: 
2016-09-08T08:03:39Z [1003] INFO [ "_jobs/7643852" ]

the log shows "_jobs/7643852" being submitted (pending state), then the job is run and fails with an assertion error. Following this the job remains in the progress state.

Basically, arangodb correctly identifies the script to be failing, but it does not move the job to the failed state (it also does not call the failure callback function). As the job remains in the progress state, the job also continues to take up a worker thread.

Any help or pointers appreciated.
Thanks

**Comment by solisoft** [comment]
A nice interface to manage queues in arangoDB web UI will be very very nice ...

**Comment by pluma** [comment]
Can you clarify whether this is a one-off issue or whether this occurs routinely?

If it's a recurring problem, does this happen with other job types as well?

**Comment by bangert** [comment]
this happens every single time. i will try to prepare a minimal test case

**Comment by bangert** [comment]
here is a foxx app exposing my problem.

https://github.com/bangert/foxx-playground/tree/master/arangodb-issue-2042

**Comment by pluma** [comment]
@bangert Can you confirm whether this has been solved in 3.1? I believe this is related to a similar problem 3.1 seems to have fixed. If that is the case we can backport the changes to 3.0 for the next bugfix release.

**Comment by dothebart** [comment]
3.0.12 is available for download via the regular upgrade channels - closing.
